java/memory/pom.xml

```diff
@@ -37,6 +37,10 @@
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
     </dependency>
+    <dependency>
```

Comment:
Is it worth taking a dependency for a single builder? At the very least, let's file a follow-up to see if we can replace any other builders in the project with Immutables as well.

Reply:
I second the idea to factor other builders. So far I didn't have a check but we may make relevant changes in other issue/pr.

And I don't think the new dependency matters much as it's a single jar library and it is not needed in runtime.

Reply:
I think this would be useful elsewhere. For example, we should really use it for the *.pojo classes. Also, as mentioned, this isn't used/required at runtime

Reply:
Ah - my mistake, non-runtime dependencies are even better.